web-ui/src/app/components/tree-visualizer/external-tree-visualizer.component.ts

```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router'
+
+@Component({
+  selector: 'grt-external-tree-visualizer',
+  templateUrl: './external-tree-visualizer.component.html',
+  styleUrls: ['./external-tree-visualizer.component.scss']
+})
+export class ExternalTreeVisualizerComponent implements OnInit {
+  public xml: string;
+  public sentence: string;
+
+  constructor(private activatedRoute: ActivatedRoute) { }
+
+  ngOnInit() {
+    this.activatedRoute.queryParams.subscribe(params => {
+      this.xml = params['XML'];
+      this.sentence = params['sent'];
+      if (!this.sentence) {
+        this.sentence = 'Unknown sentence';
```

Comment:
This would be an issue if the application would be localized (because localization is only available in templates).